dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,770 @@
+    private readonly VectorStoreRecordPropertyReader _propertyReader;
+
+    /// <summary>Flag which indicates whether vector properties exist in the consumer data model.</summary>
+    private readonly bool _vectorPropertiesExist;
+
+    /// <summary>Collection of properties to operate in SQLite data table.</summary>
+    private readonly Lazy<List<VectorStoreRecordProperty>> _dataTableProperties;
+
+    /// <summary>Collection of properties to operate in SQLite vector table.</summary>
+    private readonly Lazy<List<VectorStoreRecordProperty>> _vectorTableProperties;
+
+    /// <summary>Collection of property names to operate in SQLite data table.</summary>
+    private readonly Lazy<List<string>> _dataTableStoragePropertyNames;
+
+    /// <summary>Collection of property names to operate in SQLite vector table.</summary>
+    private readonly Lazy<List<string>> _vectorTableStoragePropertyNames;
+
+    /// <summary>Table name in SQLite for data properties.</summary>
+    private readonly string _dataTableName;
+
+    /// <summary>Table name in SQLite for vector properties.</summary>
+    private readonly string _vectorTableName;
+
+    /// <inheritdoc />
+    public string CollectionName { get; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqliteVectorStoreRecordCollection{TRecord}"/> class.
+    /// </summary>
+    /// <param name="connection"><see cref="DbConnection"/> that will be used to manage the data in SQLite.</param>
+    /// <param name="collectionName">The name of the collection/table that this <see cref="SqliteVectorStoreRecordCollection{TRecord}"/> will access.</param>
+    /// <param name="options">Optional configuration options for this class.</param>
+    public SqliteVectorStoreRecordCollection(
+        DbConnection connection,
+        string collectionName,
+        SqliteVectorStoreRecordCollectionOptions<TRecord>? options = default)
+    {
+        // Verify.
+        Verify.NotNull(connection);
+        Verify.NotNullOrWhiteSpace(collectionName);
+        VectorStoreRecordPropertyVerification.VerifyGenericDataModelKeyType(typeof(TRecord), options?.DictionaryCustomMapper is not null, SqliteConstants.SupportedKeyTypes);
+        VectorStoreRecordPropertyVerification.VerifyGenericDataModelDefinitionSupplied(typeof(TRecord), options?.VectorStoreRecordDefinition is not null);
+
+        // Assign.
+        this._connection = connection;
+        this.CollectionName = collectionName;
+        this._options = options ?? new();
+
+        this._dataTableName = this.CollectionName;
+        this._vectorTableName = GetVectorTableName(this._dataTableName, this._options);
+
+        this._propertyReader = new VectorStoreRecordPropertyReader(typeof(TRecord), this._options.VectorStoreRecordDefinition, new()
+        {
+            RequiresAtLeastOneVector = false,
+            SupportsMultipleKeys = false,
+            SupportsMultipleVectors = true
+        });
```

Comment:
I don't think so, because it depends on capabilities of actual database and the connector implementation. For example, if we add a support for multiple keys in the future for SQLite connector, we will have to change `SupportsMultipleKeys` to `true` and add additional code to support this. So, even if we expose these flags to the users, switching them won't change anything.